### PR TITLE
[2.x] Add account deletion feature

### PR DESCRIPTION
**Pull request by redidizzy**
I'm building an app that shouldn't even allow users to delete their accounts.
I thought it would be useful to add it as a feature.

### CODE REVIEW

File: routes/inertia.php [hunk 1]
```diff
@@ -16,16 +16,18 @@
     Route::group(['middleware' => ['auth', 'verified']], function () {
         // User & Profile...
         Route::get('/user/profile', [UserProfileController::class, 'show'])
-                    ->name('profile.show');
+            ->name('profile.show');
```
Comment (driesvints) [inline routes/inertia.php:19]: Please undo all styling changes in this PR.

File: stubs/livewire/resources/views/profile/show.blade.php [hunk 2]
```diff
@@ -33,11 +33,13 @@
                 @livewire('profile.logout-other-browser-sessions-form')
             </div>
 
-            <x-jet-section-border />
+            @if(Laravel\Jetstream\Jetstream::hasAccountDeletionFeatures())
```
Comment (bonzai) [inline stubs/livewire/resources/views/profile/show.blade.php:36]:
```suggestion
            @if (Laravel\Jetstream\Jetstream::hasAccountDeletionFeatures())
```

File: src/Features.php [hunk 3]
```diff
@@ -111,4 +120,14 @@ public static function teams(array $options = [])
...
     }
+
+    /**
+     * Enable the account deletion feature.
+     *
+     * @return string
+     */
+    public static function accountDeletion()
+    {
+        return 'accountDeletion';
```
Comment (bonzai) [inline src/Features.php:131]:
```suggestion
        return 'account-deletion';
```

File: src/Features.php [hunk 4]
```diff
@@ -77,6 +77,15 @@ public static function sendsTeamInvitations()
         return static::optionEnabled(static::teams(), 'invitations');
     }
 
+    /** Determine if the application is using any account deletion features.
+     *
```
Comment (bonzai) [inline src/Features.php:81]:
```suggestion
    /**
     * Determine if the application is using any account deletion features.
     *
```

**Pull request merged**